src/main/java/ee/tuleva/onboarding/mandate/content/HtmlMandateContentCreator.java

```diff
@@ -1,25 +1,25 @@
 package ee.tuleva.onboarding.mandate.content;
 
+import ee.tuleva.onboarding.config.TemplateEngineWrapper;
 import ee.tuleva.onboarding.fund.Fund;
 import ee.tuleva.onboarding.mandate.FundTransferExchange;
 import ee.tuleva.onboarding.mandate.Mandate;
 import ee.tuleva.onboarding.mandate.content.thymeleaf.ContextBuilder;
 import ee.tuleva.onboarding.user.User;
 import ee.tuleva.onboarding.user.preferences.UserPreferences;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
-import org.thymeleaf.TemplateEngine;
 import org.thymeleaf.context.Context;
-import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver;
 
-import javax.annotation.PostConstruct;
 import java.math.BigDecimal;
 import java.util.*;
 import java.util.stream.Collectors;
 
 @Component
+@RequiredArgsConstructor
 public class HtmlMandateContentCreator implements MandateContentCreator {
 
-    TemplateEngine templateEngine;
+    private final TemplateEngineWrapper templateEngine;
     User user;
```

Comment:
Holding state in a singleton Spring bean. This is not thread safe! You should design your beans stateless in order to guarantee thread safety. Read more: https://stackoverflow.com/a/15747339